### PR TITLE
Set Max Limit on number of executors to be launched in spark plugin

### PR DESCRIPTION
**Pull request by migueltol22**
# TL;DR
Allows configuration to set a max limit on spark executor by specifying `spark.executor.instances:<max_limit>` as part of the default spark configuration.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/366

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go/tasks/plugins/k8s/spark/spark.go [hunk 1]
```diff
@@ -71,6 +72,33 @@ func validateSparkJob(sparkJob *plugins.SparkJob) error {
...
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func populateSparkConfig(sparkConfig, userSparkConfig map[string]string) error {
+	for k, v := range userSparkConfig {
+		if k == "spark.executor.instances" && sparkConfig["spark.executor.instances"] != "" {
```
Comment (akhurana001) [inline go/tasks/plugins/k8s/spark/spark.go:84]: The current behavior of the config value is to be the default value if the users don't provide one. This will change its behavior as the max.

 Let's add a a separate section for limits ?

Comment (migueltol22) [reply go/tasks/plugins/k8s/spark/spark.go]: Updated with separate section.

File: go/tasks/plugins/k8s/spark/spark.go [hunk 2]
```diff
@@ -30,13 +31,20 @@ import (
...
 // Spark-specific configs
 type Config struct {
 	DefaultSparkConfig    map[string]string `json:"spark-config-default" pflag:",Key value pairs of default spark configuration that should be applied to every SparkJob"`
 	SparkHistoryServerURL string            `json:"spark-history-server-url" pflag:",URL for SparkHistory Server that each job will publish the execution history to."`
+	SparkLimits           Limits            `json:"spark-limits" pflag:", Spark Limits that can be set for each job execution."`
+}
+
+// Spark Limits config
+type Limits struct {
+	ExecutorLimit string `json:"executor-limit" pflag:",Max limit on number of executors allowed to be launched."`
```
Comment (akhurana001) [inline go/tasks/plugins/k8s/spark/spark.go:47]: nit: would suggest renaming to ExecutorCountLimit or similar just to be more explicit in-case  if we want to add memory or cpu limits as well.

**Comment by akhurana001** [comment]
looks good.

File: go/tasks/plugins/k8s/spark/spark.go (hunk 2 above)
Comment (migueltol22) [reply go/tasks/plugins/k8s/spark/spark.go]: Updated.

**Comment by kumare3** [comment]
@migueltol22 is this required and are you merging it?
